src/main/java/com/amazonaws/samples/S3Sample.java

```diff
@@ -0,0 +1,215 @@
+             * many objects might truncate their results when listing their objects,
+             * so be sure to check if the returned object listing is truncated, and
+             * use the AmazonS3.listNextBatchOfObjects(...) operation to retrieve
+             * additional results.
+             */
+            System.out.println("Listing objects");
+            ObjectListing objectListing = s3.listObjects(new ListObjectsRequest()
+                    .withBucketName(bucketName)
+                    .withPrefix("My"));
+            for (S3ObjectSummary objectSummary : objectListing.getObjectSummaries()) {
+                System.out.println(" - " + objectSummary.getKey() + "  " +
+                        "(size = " + objectSummary.getSize() + ")");
+            }
+            System.out.println();
+
+            /*
+             * Delete an object - Unless versioning has been turned on for your bucket,
+             * there is no way to undelete an object, so use caution when deleting objects.
+             */
+            System.out.println("Deleting an object\n");
+            s3.deleteObject(bucketName, key);
+
+            /*
+             * Delete a bucket - A bucket must be completely empty before it can be
+             * deleted, so remember to delete any objects from your buckets before
+             * you try to delete them.
+             */
+            System.out.println("Deleting bucket " + bucketName + "\n");
+            s3.deleteBucket(bucketName);
+        } catch (AmazonServiceException ase) {
+            System.out.println("Caught an AmazonServiceException, which means your request made it "
+                    + "to Amazon S3, but was rejected with an error response for some reason.");
+            System.out.println("Error Message:    " + ase.getMessage());
+            System.out.println("HTTP Status Code: " + ase.getStatusCode());
+            System.out.println("AWS Error Code:   " + ase.getErrorCode());
+            System.out.println("Error Type:       " + ase.getErrorType());
+            System.out.println("Request ID:       " + ase.getRequestId());
+        } catch (AmazonClientException ace) {
+            System.out.println("Caught an AmazonClientException, which means the client encountered "
+                    + "a serious internal problem while trying to communicate with S3, "
+                    + "such as not being able to access the network.");
+            System.out.println("Error Message: " + ace.getMessage());
+        }
+    }
+
+    /**
+     * Creates a temporary file with text data to demonstrate uploading a file
+     * to Amazon S3
+     *
+     * @return A newly created temporary file with text data.
+     *
+     * @throws IOException
+     */
+    private static File createSampleFile() throws IOException {
+        File file = File.createTempFile("aws-java-sdk-", ".txt");
+        file.deleteOnExit();
+
+        Writer writer = new OutputStreamWriter(new FileOutputStream(file));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer

**Problem**
This line of code contains a resource that might not be closed properly. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *writer*. The resource is referenced in statements at the following lines: 185, 186, 187, 188, 189, 190. The resource closure statement is at line: 191. There are other execution paths that do not contain closure statements, e.g., when exception is thrown by Writer.write. Either a) close *writer* in a try-finally block or b) close the resource by declaring *writer* in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).